### DISCUSSION TITLE
What happens when you delete a xstate/store?

### DISCUSSION THREAD

**Discussion by ping-maxwell** [statelyai/xstate#5161] [Q&A]
I want to have a global object called `windows`. Within, contains keys representing the window id, and the values would be entirely new stores created via `createStore`. 

My first question is: 
is this fine to do so? I don't like the idea of having an array which then contains all the windows with that all being managed by 1 store since something like `useSelector` and other things may have issues dealing with arrays and objs within.

My second question is:
What happens if I call `delete` on a key in the windows global object? Does xstate/store still run things in the background not knowing that I just deleted the store or will it safely not do anything?

**Comment by davidkpiano** [comment]
It is strongly recommended that you only do immutable updates. Would this work for your use-case?
